custom_components/waste_collection_schedule/waste_collection_schedule/source/samiljo_se.py

```diff
@@ -125,37 +130,73 @@ def fetch(self):
-                            Collection(
-                                t=t,
-                                icon=icon,
-                                date=datetime.date(year, month, day),                       
-                            )
-                    )
+        # Extended list of bin types to cover all possible collection types
+        bin_types = [
+            'HKARL1', 'HKARL1-H', 'HKARL2', 'HKARL2-H',
+            'FKARL1', 'FKARL1-H', 'FKARL2', 'FKARL2-H', 
+            'HMAT', 'HMAT-H', 'HREST', 'HREST-H',
+            'HOSORT', 'HOSORT-H', 'FOSORT', 'FOSORT-H',
+            'HREST-HK', 'HREST-HK-H', 'HKARL1-HK', 'HKARL1-HK-H',
+            'TRG', 'TRG-H', 'FREST-HK', 'FREST-HK-H',
+            'FKARL1-HK', 'FKARL1-HK-H', 'FREST', 'FREST-H'
+        ]
+        
+        for bin_type in bin_types:
+            elements = soup.find_all("td", class_=bin_type)
+            
+            for element in elements:
+                # Find the day cell containing this bin collection
+                day_cell = element
+                steps = 0
+                while day_cell and steps < 10:  # Prevent infinite loop
+                    day_cell = day_cell.parent
+                    steps += 1
+                    
+                    if day_cell and day_cell.name == 'td' and day_cell.get('class'):
+                        day_classes = day_cell.get('class')
+                        
+                        # Check if this is a day cell
+                        if any('styleDay' in cls for cls in day_classes):
+                            break
+                            
+                    if day_cell and day_cell.name == 'body':
+                        day_cell = None
+                        break
+                
+                if not day_cell:
+                    continue
+                
+                # Find day number
+                day_element = day_cell.find("div", ["styleInteIdag", "styleIdag"])
+                if day_element:
+                    day_text = day_element.get_text().strip()
+                    if day_text.isdigit():
+                        day = int(day_text)
+                        
+                        # Find month
+                        month_element = day_cell.find_previous("td", "styleMonthName")
+                        if month_element:
+                            month_text = month_element.get_text().strip()
+                            
+                            # Parse month and year
+                            if '-' in month_text:
+                                month_part, year_part = month_text.split('-', 1)
+                                month_name = month_part.strip().lower()
+                                year = int(year_part.strip())
+                                
+                                if month_name in MONTH_MAP:
+                                    month = MONTH_MAP[month_name]
+                                    
```

Comment:
I peronally don' like that deep of an indentaiton leven. It makes it hard to read. I perefer 

```py
if not day_elelent:
  continue
...
if not day_text.isdigit():
  continue
...
if not month_element:
  continue
...
if '-' not in month_text:
  continue
....
if month_name not in MONTH_MAP:
  continue
....
```